src/common/src/types/mod.rs

```diff
@@ -241,6 +241,32 @@ impl DataType {
             List { datatype } => datatype.mem_cmp_eq_value_enc(),
         }
     }
+
+    pub fn get_fixed_len(&self) -> Option<usize> {
```

Comment:
IMHO, this function looks like `encoding_data_size`.

shall we provide a function to check if `data_type` and  use the `encoding_data_size` function uniformly ?